storage/country.cpp

```diff
@@ -19,99 +19,105 @@ namespace storage
 using TMwmSubtreeAttrs = pair<uint32_t, size_t>;
 
 template <class ToDo>
-TMwmSubtreeAttrs LoadGroupSingleMwmsImpl(int depth, json_t * group, TCountryId const & parent, ToDo & toDo)
+TMwmSubtreeAttrs LoadGroupSingleMwmsImpl(int depth, json_t * node, TCountryId const & parent, ToDo & toDo)
 {
   uint32_t mwmCounter = 0;
   size_t mwmSize = 0;
-  size_t const groupListSize = json_array_size(group);
-  for (size_t i = 0; i < groupListSize; ++i)
-  {
-    json_t * j = json_array_get(group, i);
-
-    char const * id = json_string_value(json_object_get(j, "id"));
-    if (!id)
-      MYTHROW(my::Json::Exception, ("LoadGroupImpl. Id is missing.", id));
 
-    uint32_t const nodeSize = static_cast<uint32_t>(json_integer_value(json_object_get(j, "s")));
-    // We expect that mwm and routing files should be less than 2GB.
-    Country * addedNode = toDo(id, nodeSize, depth, parent);
+  char const * id = json_string_value(json_object_get(node, "id"));
+  if (!id)
+    MYTHROW(my::Json::Exception, ("LoadGroupImpl. Id is missing.", id));
 
-    json_t * oldIds = json_object_get(j, "old");
-    if (oldIds)
+  // Mapping two component (big) mwms to one componenst (small) ones.
+  json_t * oldIds = json_object_get(node, "old");
+  if (oldIds)
+  {
+    size_t const oldListSize = json_array_size(oldIds);
+    for (size_t k = 0; k < oldListSize; ++k)
     {
-      size_t const oldListSize = json_array_size(oldIds);
-      for (size_t k = 0; k < oldListSize; ++k)
-      {
-        string oldIdValue = json_string_value(json_array_get(oldIds, k));
-        toDo(oldIdValue, id);
-      }
+      string oldIdValue = json_string_value(json_array_get(oldIds, k));
+      toDo.InsertOldMwmMapping(id, oldIdValue);
```

Comment:
toDo is used as a callable object in this method as well (doStore(id, nodeSize, depth, parent)). I renamed toDo to doStore BTW.

> Probably it's worth get rid of template parameter and rename toDo to something more meaningful.

I thought about it but it's not easy now. LoadGroupSingleMwmsImpl is used with two different classes with the same interface.